### PR TITLE
[WIP-Do not merge] Refactoring and cleanup of BG transform registers

### PR DESCRIPTION
**Pull request by negativeExponent**
**WIP - DO NOT MERGE**
This has not been tested with the threading feature or on vita which the optimizations were done. Even though i can compile threading support on linux, i cannot confirm how it works since core crashes trying to load a game (it works on debug build but there is no video, its just black screen with sound running in background).
This has to be tested on vita and see if it has negative effects. This affects bg rotations ingame or scaling: Some games to test includes:
- Pinball Tycoon for that scoreboard bug
- Ace Lightning - the intro has at least 2 screens with rotating logos
- F-Zero Climax - the race track itself uses this.

####

- Basically, moves per-scanline checks/computations to actual
  io register writes.
- Fixes lingering scanline timing that affects Pinball Tycoon's
  scoreboard
- Moves some registers into io_register array

### CODE REVIEW

**Comment by negativeExponent** [comment]
Closing this since i have no means to continue it.
https://github.com/libretro/vba-next/issues/137#issuecomment-691598792
